library/alloc/src/slice.rs

```diff
@@ -222,30 +235,19 @@ impl<T> [T] {
         stable_sort(self, T::lt);
     }
 
-    /// Sorts the slice with a comparator function, preserving initial order of equal elements.
+    /// Sorts the slice with a comparison function, preserving initial order of equal elements.
     ///
     /// This sort is stable (i.e., does not reorder equal elements) and *O*(*n* \* log(*n*))
     /// worst-case.
     ///
-    /// The comparator function should define a total ordering for the elements in the slice. If the
-    /// ordering is not total, the order of the elements is unspecified.
-    ///
-    /// If the comparator function does not implement a total order the resulting order is
-    /// unspecified. All original elements will remain in the slice and any possible modifications
-    /// via interior mutability are observed in the input. Same is true if the comparator function
-    /// panics. A total order (for all `a`, `b` and `c`):
+    /// If the comparison function `compare` does not implement a [total order] the resulting order
+    /// of elements in the slice is unspecified. All original elements will remain in the slice and
+    /// any possible modifications via interior mutability are observed in the input. Same is true
+    /// if `compare` panics.
```

Comment:
(non-blocking)
Same internal mutability Q.

This should maybe should be even clearer about which inputs may exhibit internal mutability given the function passed to `sort_by` is FnMut? But explaining the relationship to the `FnMut` thing isn't necessary given that's a preexisting concern.

Reply:
I touched on why I think it wouldn't be a bad idea to mention it here https://github.com/rust-lang/rust/pull/128273#discussion_r1702748863. No strong feeling either way. It's a non-trivial property that I'd argue is not obvious even if you know what `FnMut` does. One needs to connect a lot of dots to figure out that the function must maintain that property to be sound. I [didn't](https://youtu.be/3JZAQ4Gsl-g?t=1047) initially. So I could see other users not realizing that this property is a given. It only costs us a sentence, and if the confusion potential is low, I'd say the benefits of mentioning it outweigh the costs.